File: meldingsvarsler-oppgave/src/main/java/no/sonat/meldingsvarsler/meldinger/sms/SMSProsessor.java
package no.sonat.meldingsvarsler.meldinger.sms;

public class SMSProsessor {

    public void sendSms(SMSAbonnent smsAbonnent, SMSMelding smsMelding) {
        System.out.println("Sender SMS til "
                + smsAbonnent.telefonnummer()
                + "(" + smsAbonnent.navn() + ")"
                + " => " + smsMelding.tekstmelding
        );
    }
}

File: meldingsvarsler-fasit-forslag/App/SMSProsessor.cs
using System;

namespace App
{
    public class SmsProsessor : IMeldingProsessor
    {
        public bool HaandtererUtsendingAv(IMelding melding)
        {
            return melding is SmsMelding;
        }

        public void SendMelding(IAbonnent abonnent, IMelding melding)
        {
            SendSms((SmsAbonnent) abonnent, (SmsMelding) melding);
        }

        public void SendSms(SmsAbonnent smsAbonnent, SmsMelding smsMelding)
        {
            Console.Out.WriteLine("Sender SMS til "
                                + smsAbonnent.TelefonNummer
                                + " (" + smsAbonnent.Navn + ")"
                                + " => " + smsMelding.Melding
            );
        }
    }
}